Errors on chat screen : Desktop

https://github.com/user-attachments/assets/2beb0476-8461-4ba1-bc52-aa9b29483ded

```
[2024-07-30][11:58:31.147162][a3::home::client_notifier][ERROR] platform dispatch error setState() or markNeedsBuild() called during build.
This AvatarBuilder widget cannot be marked as needing to build because the framework is already in the process of building widgets. A widget can be marked as needing to be built during the build phase only if one of its ancestors is currently building. This exception is allowed because the framework builds parent widgets before children, which means a dirty descendant will always be built. Otherwise, the framework might not visit this widget during this build phase.
The widget on which setState() or markNeedsBuild() was called was:
  AvatarBuilder
The widget which was currently being built when the offending call was made was:
  UncontrolledProviderScope null #0      Element.markNeedsBuild.<anonymous closure> (package:flutter/src/widgets/framework.dart:5047:9)
#1      Element.markNeedsBuild (package:flutter/src/widgets/framework.dart:5059:6)
#2      ConsumerStatefulElement.watch.<anonymous closure>.<anonymous closure> (package:flutter_riverpod/src/consumer.dart:566:20)
#3      ProviderBase.addListener.<anonymous closure> (package:riverpod/src/framework/provider_base.dart:98:41)
#4      _RootZone.runBinaryGuarded (dart:async/zone.dart:1606:10)
#5      ProviderElementBase._notifyListeners.<anonymous closure> (package:riverpod/src/framework/element.dart:525:28)
#6      ResultData.map (package:riverpod/src/result.dart:74:16)
#7      ProviderElementBase._notifyListeners (package:riverpod/src/framework/element.dart:519:14)
#8      ProviderElementBase._performBuild (package:riverpod/src/framework/element.dart:358:7)
#9      ProviderElementBase.flush (package:riverpod/src/framework/element.dart:307:7)
#10     ProviderScheduler._performRefresh (package:riverpod/src/framework/scheduler.dart:94:41)
#11     ProviderScheduler._task (package:riverpod/src/framework/scheduler.dart:82:5)
#12     ProviderScheduler.vsync.<anonymous closure>.invoke (package:riverpod/src/framework/scheduler.dart:41:15)
#13     _UncontrolledProviderScopeElement.build (package:flutter_riverpod/src/framework.dart:396:12)
#14     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:5487:15)
#15     Element.rebuild (package:flutter/src/widgets/framework.dart:5203:7)
#16     BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2905:19)
#17     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:1136:21)
#18     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:443:5)
#19     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1392:15)
#20     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1313:9)
#21     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1171:5)
#22     _invoke (dart:ui/hooks.dart:312:13)
#23     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:419:5)
#24     _drawFrame (dart:ui/hooks.dart:283:31)


======== Exception caught by scheduler library =====================================================
The following assertion was thrown during a scheduler callback:
'package:flutter/src/rendering/mouse_tracker.dart': Failed assertion: line 200 pos 12: '!_debugDuringDeviceUpdate': is not true.


Either the assertion indicates an error in the framework itself, or we should provide substantially more information in this error message to help you determine and fix the underlying cause.
In either case, please report this assertion by filing a bug on GitHub:
  https://github.com/flutter/flutter/issues/new?template=2_bug.yml

When the exception was thrown, this was the stack: 
#2      MouseTracker._deviceUpdatePhase (package:flutter/src/rendering/mouse_tracker.dart:200:12)
#3      MouseTracker.updateAllDevices (package:flutter/src/rendering/mouse_tracker.dart:367:5)
#4      RendererBinding._scheduleMouseTrackerUpdate.<anonymous closure> (package:flutter/src/rendering/binding.dart:460:22)
#5      SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1392:15)
#6      SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1326:11)
#7      SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1171:5)
#8      _invoke (dart:ui/hooks.dart:312:13)
#9      PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:419:5)
#10     _drawFrame (dart:ui/hooks.dart:283:31)
(elided 2 frames from class _AssertionError)
====================================================================================================
```

--- a/app/lib/features/chat/providers/notifiers/chat_room_notifier.dart
+++ b/app/lib/features/chat/providers/notifiers/chat_room_notifier.dart
@@ -227,7 +227,14 @@ class ChatRoomNotifier extends StateNotifier<ChatRoomState> {
 
   // fetch original content media for reply msg, i.e. text/image/file etc.
   Future<void> fetchOriginalContent(String originalId, String replyId) async {
-    final roomMsg = await timeline.getMessage(originalId);
+    RoomMessage roomMsg;
+    try {
+      roomMsg = await timeline.getMessage(originalId);
+    } catch (error, stack) {
+      _log.severe('Failing to load reference $replyId (from $originalId)',
+          error, stack,);
+      return;
+    }
 
     // reply is allowed for only EventItem not VirtualItem
     // user should be able to get original event as RoomMessage

--- a/app/lib/features/chat/widgets/chats_list.dart
+++ b/app/lib/features/chat/widgets/chats_list.dart
@@ -134,25 +134,27 @@ class __AnimatedChatsListState extends State<_AnimatedChatsList> {
   }
 
   void refreshList() {
-    final diffResult = calculateListDiff<String>(
-      _currentList,
-      widget.entries,
-      detectMoves: false,
-    );
-    for (final update in diffResult.getUpdatesWithData()) {
-      update.when(
-        insert: _insert,
-        remove: _remove,
-        change: (pos, oldData, newData) {
-          _remove(pos, oldData);
-          _insert(pos, newData);
-        },
-        move: (from, to, item) {
-          _remove(from, item);
-          _insert(to, item);
-        },
+    WidgetsBinding.instance.addPostFrameCallback((Duration duration) {
+      final diffResult = calculateListDiff<String>(
+        _currentList,
+        widget.entries,
+        detectMoves: false,
       );
-    }
+      for (final update in diffResult.getUpdatesWithData()) {
+        update.when(
+          insert: _insert,
+          remove: _remove,
+          change: (pos, oldData, newData) {
+            _remove(pos, oldData);
+            _insert(pos, newData);
+          },
+          move: (from, to, item) {
+            _remove(from, item);
+            _insert(to, item);
+          },
+        );
+      }
+    });
   }
 
   void _insert(int pos, String data) {

